Fuzzy search and related endpoints

lindseymazurowski opened the pull request.

lindseymazurowski committed:
fuzzy simple search deployment

lindseymazurowski committed:
Update fuzzysearch.py

lindseymazurowski committed:
advanced item search endpoint

lindseymazurowski committed:
add function to create search terms for designated entries

lindseymazurowski committed:
Updated parser to create search terms after new entries are inserted

KingJMS1 reviewed (changes requested):
Precompiled binaries are not available for for fuzzy for python 3.10. I need these in order to install it on the server. Either compile me a version for 3.10 or use a different method.

lindseymazurowski committed:
updated fuzzy search with new library

removed defunct "fuzzy" library and replaced with "jellyfish" library (https://pypi.org/project/jellyfish/) which has similar functionality but is up to date with python 3.10

lindseymazurowski committed:
Split advanced search into fuzzy and specific

KingJMS1 reviewed (changes requested):
The fuzzy search is not partial, i.e. if I search rob, results for robert do not show up even though they should.

lindseymazurowski committed:
Update fuzzy searches to include partial matching

KingJMS1 reviewed (approved):
Good, will deploy changes now.
